File: benchmarks/benchmark-micronaut-data-jpa/src/main/java/example/ProductRepository.java
package example;

import io.micronaut.data.annotation.*;
import io.micronaut.data.repository.CrudRepository;
import java.util.List;

@Repository
public interface ProductRepository extends CrudRepository<Product, Long> {

    @Join("manufacturer") // <1>
    List<Product> list();
}

File: doc-examples/hibernate-reactive-example-kotlin/src/main/kotlin/example/ProductRepository.kt
package example

import io.micronaut.data.annotation.*
import io.micronaut.data.jpa.annotation.EntityGraph
import io.micronaut.data.repository.kotlin.CoroutineCrudRepository
import kotlinx.coroutines.flow.Flow

// tag::join[]
// tag::async[]
@Repository
interface ProductRepository : CoroutineCrudRepository<Product, Long> {
// end::join[]
// end::async[]

    // tag::join[]
    @Join(value = "manufacturer", type = Join.Type.FETCH) // <1>
    suspend fun list(): List<Product>
    // end::join[]

    // tag::entitygraph[]
    @EntityGraph(attributePaths = ["manufacturer", "title"]) // <1>
    override fun findAll(): Flow<Product>
    // end::entitygraph[]


// tag::join[]
// tag::async[]
}
// end::join[]
// end::async[]
